packages/aws-cdk-lib/aws-batch/lib/ecs-container-definition.ts

```diff
@@ -1084,9 +1098,16 @@ export class EcsFargateContainerDefinition extends EcsContainerDefinitionBase im
       },
       runtimePlatform: {
         cpuArchitecture: this.fargateCpuArchitecture?._cpuArchitecture,
-        operatingSystemFamily: this.fargateOperatingSystemFamily?._operatingSystemFamily,
+        operatingSystemFamily,
       },
     };
+
+    // readonlyRootFilesystem isn't applicable to Windows, see https://kubernetes.io/docs/concepts/windows/intro/
+    if (this.isWindows(operatingSystemFamily)) {
+      containerDef.readonlyRootFilesystem = undefined;
+    }
```

Comment:
@pahud `operatingSystemFamily` isn't available in the constructor so I had to remove it here from the final rendering.